cfgov/unprocessed/js/molecules/Multiselect.js

```diff
@@ -250,50 +240,53 @@ function Multiselect( element ) { // eslint-disable-line max-statements, inline-
    * @param   {string} value The value of the option the user has chosen
    */
   function _updateSelections( value ) {
-    var optionIndex = arrayHelpers.indexOfObject( _optionData, 'value', value );
-    var option = _optionData[optionIndex] || _optionData[_index];
+    var optionIndex
+      = arrayHelpers.indexOfObject( _optionsData, 'value', value );
+    var option = _optionsData[optionIndex] || _optionsData[_index];
 
     if ( option ) {
-      var selectionIndex = _selections.indexOf( option );
-      var li;
+      var _selectionsItemDom;
 
-      if ( selectionIndex > -1 ) {
-        _selections.splice( selectionIndex, 1 );
-
-        if ( _list.classList.contains( 'max-selections' ) ) {
-          _list.classList.remove( 'max-selections' );
+      if ( option.checked ) {
+        if ( _optionsDom.classList.contains( 'max-selections' ) ) {
+          _optionsDom.classList.remove( 'max-selections' );
         }
 
-        li = _choices.querySelector( 'li[data-option="' + option.value + '"]' );
+        _selectionsItemDom =_selectionsDom
+          .querySelector( '[data-option="' + option.value + '"]' );
 
-        if ( li ) {
-          _choices.removeChild( li );
+        if ( _selectionsItemDom ) {
+          _selectionsDom.removeChild( _selectionsItemDom );
         }
+        option.checked = false;
+        _selectionsCount -= 1;
       } else {
-        li = domCreate( 'li', {
+        _selectionsItemDom = domCreate( 'li', {
           'data-option': option.value
         } );
 
         domCreate( 'label', {
-          'for': option.value,
+          'for':       option.value,
           'innerHTML': option.text,
-          'inside': li
+          'inside':    _selectionsItemDom
         } );
 
-        _choices.appendChild( li );
-        _selections.push( option );
+        _selectionsDom.appendChild( _selectionsItemDom );
 
-        if ( _selections.length > 4 ) {
-          _list.classList.add( 'max-selections' );
+        option.checked = true;
+        _selectionsCount += 1;
+
+        if ( _selectionsCount > 4 ) {
```

Comment:
It could be helpful to move `4` to a constant at the top of the class. `var MAX_SELECTIONS = 4;` or something.